rct229/rulesets/ashrae9012019/section5/section5rule1.py

```diff
@@ -0,0 +1,255 @@
+                )
+
+            def get_calc_vals(self, context, data=None):
+                building_b0 = context.BASELINE_0
+                building_b90 = context.BASELINE_90
+                building_b180 = context.BASELINE_180
+                building_b270 = context.BASELINE_270
+                building_p = context.PROPOSED
+                building_u = context.USER
+
+                baseline_list = data["baseline_list"]
+                no_of_rmds = data["no_of_rmds"]
+                no_of_output_instance = data["no_of_output_instance"]
+
+                has_proposed = find_one("$.type", building_p, False)
+                has_user = find_one("$.type", building_u, False)
+                has_baseline_0 = RULESET_MODEL.BASELINE_0 in baseline_list
+                has_baseline_90 = RULESET_MODEL.BASELINE_90 in baseline_list
+                has_baseline_180 = RULESET_MODEL.BASELINE_180 in baseline_list
+                has_baseline_270 = RULESET_MODEL.BASELINE_270 in baseline_list
+
+                has_proposed_output = has_proposed and find_one(
+                    "$.output.output_instance", building_p, False
+                )
+                has_user_output = has_user and find_one(
+                    "$.output.output_instance", building_u, False
+                )
+                has_basseline_0_output = has_baseline_0 and find_one(
+                    "$.output.output_instance", building_b0, False
+                )
+                has_basseline_90_output = has_baseline_90 and find_one(
+                    "$.output.output_instance", building_b90, False
+                )
+                has_basseline_180_output = has_baseline_180 and find_one(
+                    "$.output.output_instance", building_b180, False
+                )
+                has_basseline_270_output = has_baseline_270 and find_one(
+                    "$.output.output_instance", building_b270, False
+                )
+
+                # define a function to find the azimuth's corresponding key
+                find_key_for_azi = lambda azi: next(
+                    (
+                        key
+                        for key, range_ in azimuth_fen_area_dict_b.items()
+                        if int(key.split("-")[0]) <= azi <= int(key.split("-")[1])
+                    ),
+                    None,
+                )
+
+                azimuth_fen_area_dict_b = {
+                    f"{azi}-{azi+3}": ZERO.AREA for azi in range(3, 360, 3)
+                }
+                total_surface_fenestration_area_b = ZERO.AREA
+                for surface_b in find_all(
+                    "$.building_segments[*].zones[*].surfaces[*]", building_b0
+                ):
+                    if get_opaque_surface_type(surface_b) == OST.ABOVE_GRADE_WALL:
+                        surface_azimuth_b = surface_b["azimuth"]
+
+                        if surface_azimuth_b not in azimuth_fen_area_dict_b:
```

Comment:
surface_azimuth_b I believe is an integer and will not be `in azimuth_fen_area_dict_b` since it was prepopulated with strings

Reply:
I was thinking that it could be good to populate the dictionary with integers in range(3, 360, 3), then round `surface_azimuth_b` to the nearest 3 degrees

Reply:
Good catch. These lines are not needed because ```ZERO.AREA``` was already set up above as a default value.